sequencer/src/bin/espresso-dev-node.rs

```diff
@@ -0,0 +1,326 @@
+use std::{io, time::Duration};
+
+use async_compatibility_layer::logging::{setup_backtrace, setup_logging};
+use async_std::task::spawn;
+use clap::Parser;
+use es_version::SEQUENCER_VERSION;
+use ethers::types::Address;
+use futures::FutureExt;
+use sequencer::{
+    api::options,
+    api::test_helpers::TestNetwork,
+    hotshot_commitment::{run_hotshot_commitment_task, CommitmentTaskOptions},
+    persistence,
+    testing::TestConfig,
+};
+use sequencer_utils::{
+    deployer::{deploy, Contract, Contracts},
+    AnvilOptions,
+};
+use tide_disco::{error::ServerError, Api};
+use url::Url;
+use vbs::version::StaticVersionType;
+
+#[derive(Clone, Debug, Parser)]
+struct Args {
+    /// A JSON-RPC endpoint for the L1 to deploy to. If this is not provided, an Avil node will be
+    /// launched automatically.
+    #[clap(short, long, env = "ESPRESSO_SEQUENCER_L1_PROVIDER")]
+    rpc_url: Option<Url>,
+    /// Mnemonic for an L1 wallet.
+    ///
+    /// This wallet is used to deploy the contracts, so the account indicated by ACCOUNT_INDEX must
+    /// be funded with with ETH.
+    #[clap(
+        long,
+        name = "MNEMONIC",
+        env = "ESPRESSO_SEQUENCER_ETH_MNEMONIC",
+        default_value = "test test test test test test test test test test test junk"
+    )]
+    mnemonic: String,
+    /// Account index in the L1 wallet generated by MNEMONIC to use when deploying the contracts.
+    #[clap(
+        long,
+        name = "ACCOUNT_INDEX",
+        env = "ESPRESSO_DEPLOYER_ACCOUNT_INDEX",
+        default_value = "0"
+    )]
+    account_index: u32,
+    /// Port that the HTTP API will use.
+    #[clap(long, env = "ESPRESSO_SEQUENCER_API_PORT")]
+    sequencer_api_port: u16,
+    /// If provided, the service will run a basic HTTP server on the given port.
+    ///
+    /// The server provides healthcheck and version endpoints.
+    #[clap(short, long, env = "ESPRESSO_COMMITMENT_TASK_PORT")]
+    commitment_task_port: u16,
```

Comment:
I think we still need to add the `builder_port` back here.

Reply:
I understand the need for this, but its going to require some modification to `HotShot`. There is a lot of work here so I think its best to merge. I've created an issue so we can deal w/ this in a follow up: https://github.com/EspressoSystems/espresso-sequencer/issues/1522